partition_version.cc

```diff
@@ -586,10 +611,9 @@ partition_snapshot::range_tombstones(position_in_partition_view start, position_
     while (!streams.empty()) {
         std::pop_heap(streams.begin(), streams.end(), stream_less);
         range_tombstone_list::iterator_range& stream = streams.back();
-        if (callback(stream.begin()->tombstone()) == stop_iteration::yes) {
+        if (callback(pop_stream(stream)) == stop_iteration::yes) {
             return stop_iteration::no;
         }
-        stream.advance_begin(1);
```

Comment:
This changes the behaviour also when `reversed == false`. Was the previous behaviour incorrect?

Reply:
stream.advance_begin(1) wasn't called when `callback(stream.begin()->tombstone()) == stop_iteration::yes` previously

Reply:
ok. I see that `streams` is a local variable so its state does not matter after the function returns.